subprojects/core/src/integTest/groovy/org/gradle/api/tasks/ExecOutputIntegrationTest.groovy

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.tasks
+
+import org.gradle.integtests.fixtures.AbstractConsoleFunctionalSpec
+import spock.lang.Issue
+import spock.util.environment.OperatingSystem
+
+class ExecOutputIntegrationTest extends AbstractConsoleFunctionalSpec {
+    private static final String EXPECTED_OUTPUT = "Hello, World!"
+
+    @Issue("https://github.com/gradle/gradle/issues/2009")
+    def "Project#javaexec output is grouped with it's task output"() {
+        given:
+        generateMainJavaFileEchoing(EXPECTED_OUTPUT)
+        buildFile << """apply plugin: 'java'
+
+task run {
+    dependsOn 'compileJava'
+    doLast {
+        project.javaexec {
+            classpath = sourceSets.main.runtimeClasspath
+            main = 'Main'
+        }
+    }
+}
+"""
+
+        when:
+        succeeds("run")
+
+        then:
+        (result.output =~ /(?ms)(> Task :run.*?$EXPECTED_OUTPUT)/).find()
```

Comment:
This logic should go in a test fixture so it can be reused by other tests. We now have a bunch of tests all implementing the same kind of logic in different ways.